RELEASES.md

```diff
@@ -1,3 +1,187 @@
+Version 1.58.0 (2022-01-13)
+==========================
+
+Language
+--------
+
+- [Format strings can now capture arguments simply by writing `{ident}` in the string.][90473]
+- [`*const T` pointers can now be dereferenced in const contexts.][89551]
+- [The rules for when a generic struct implements `Unsize` have been relaxed.][90417]
+
+Compiler
+--------
+
+- [Add LLVM CFI support to the Rust compiler][89652]
+- [Stabilize -Z strip as -C strip][90058]
+- [Add support for LLVM coverage mapping format versions 5 and 6][91207]
+- [Emit LLVM optimization remarks when enabled with `-Cremark`][90833]
+- [Update the minimum external LLVM to 12][90175]
+- [Add `x86_64-unknown-none` at Tier 3*][89062]
+- [Build musl dist artifacts with debuginfo enabled][90733]
+- [Don't abort compilation after giving a lint error][87337]
+- [Error messages point at the source of trait bound obligations in more places][89580]
+
+\* Refer to Rust's [platform support page][platform-support-doc] for more
+   information on Rust's tiered platform support.
+
+Libraries
+---------
+
+- [Add #[must_use] to Rc::downgrade][89833]
+- [Add #[must_use] to expensive computations][89835]
+- [Add #[must_use] to mem/ptr functions][89839]
+- [Add #[must_use] to remaining core functions][89897]
+- [Add #[must_use] to remaining alloc functions][89899]
+- [Add #[must_use] to len and is_empty][89786]
+- [Add #[must_use] to thread::Builder][89789]
+- [Add #[must_use] to alloc functions that would leak memory][90427]
+- [Add #[must_use] to remaining std functions (A-N)][90430]
+- [Add #[must_use] to remaining std functions (O-Z)][90431]
```

Comment:
```suggestion
- [All remaining functions in the standard library have `#[must_use]` annotations where appropriate][89692], producing a warning when ignoring their return value. This helps catch mistakes such as expecting a function to mutate a value in place rather than return a new value.
```